src/pybind/mgr/dashboard/frontend/src/app/shared/components/usage-bar/usage-bar.component.ts

```diff
@@ -1,53 +1,174 @@
-import { Component, Input, OnChanges } from '@angular/core';
-
-import _ from 'lodash';
+import { Component, Input, ElementRef, OnInit, TemplateRef } from '@angular/core';
+import { ChartTabularData, MeterChartOptions, Statuses } from '@carbon/charts-angular';
+import { StatusToCssMap } from '../../enum/usage-bar-chart.enum';
+import { CssHelper } from '../../classes/css-helper';
+import { DimlessBinaryPipe } from '../../pipes/dimless-binary.pipe';
+import { DimlessPipe } from '../../pipes/dimless.pipe';
 
 @Component({
   selector: 'cd-usage-bar',
   templateUrl: './usage-bar.component.html',
   styleUrls: ['./usage-bar.component.scss']
 })
-export class UsageBarComponent implements OnChanges {
-  @Input()
-  total: number;
-  @Input()
-  used: any;
-  @Input()
-  warningThreshold?: number;
-  @Input()
-  errorThreshold?: number;
-  @Input()
-  isBinary = true;
-  @Input()
-  decimals = 0;
-  @Input()
-  calculatePerc = true;
-  @Input()
-  title = $localize`usage`;
-  @Input()
-  customLegend?: string;
-  @Input()
-  customLegendValue?: string;
-  @Input()
-  showFreeToolTip = true;
-  @Input()
-  showMultisiteTooltip = false;
-
-  usedPercentage: number;
-  freePercentage: number;
-
-  ngOnChanges() {
-    if (this.calculatePerc) {
-      this.usedPercentage = this.total > 0 ? (this.used / this.total) * 100 : 0;
-      this.freePercentage = 100 - this.usedPercentage;
+export class UsageBarComponent implements OnInit {
+  @Input({ required: true }) total: number = 0; // in bytes
+  @Input({ required: true }) used: number = 0; // in bytes
+  @Input() title: string = '';
+  @Input() warningThreshold: number = 0.9;
+  @Input() errorThreshold: number = 0.95;
+  @Input() isBinary: boolean = true;
+  @Input() decimals: number = 2;
+  @Input() proportional: boolean = true;
+
+  @Input() height: string = '2rem';
+  @Input() width: string = '15rem';
+  @Input() totalUnit: string = 'B'; // Bytes
+  @Input() animations: boolean = false;
+
+  @Input() legendEnabled: boolean = false;
+  @Input() toolbarEnabled: boolean = false;
+  @Input() noDataText: string = '-';
+
+  @Input() tooltipOptions?: MeterChartOptions['tooltip'] = {};
+  @Input() customLabel: TemplateRef<any> | string = '';
+  @Input() customColorScale?: { [key: string]: string };
+  @Input() customTotalFormatter?: (total: number) => string;
+  @Input() customBreakdownFormatter?: (x: { used: number; total: number }) => string;
```

Comment:
whats this 'spacing' separation for ?  can we add  a comment if there is any logical separation ?
if not lets remove it, and do any logical separation that makes sense.

This is a generic component so add comments again each property about its usage.

Thanks